libafl_targets/src/sancov_cmp.c

```diff
@@ -8,60 +8,48 @@
   #include "cmplog.h"
 #endif
 
-void __sanitizer_cov_trace_cmp1(uint8_t arg1, uint8_t arg2) {
-  uintptr_t k = RETADDR;
-  k = (k >> 4) ^ (k << 8);
-
-#ifdef SANCOV_VALUE_PROFILE
-  k &= CMP_MAP_SIZE - 1;
-  __libafl_targets_value_profile1(k, arg1, arg2);
-#endif
-#ifdef SANCOV_CMPLOG
-  k &= CMPLOG_MAP_W - 1;
-  cmplog_instructions_checked(k, 1, (uint64_t)arg1, (uint64_t)arg2);
+// Note: for RETADDR to give us the fuzz target caller address we need 
+//       to guarantee that this code is inlined. `inline` keyword provides
+//       no such guarantees, but a macro does.
+#if defined(SANCOV_VALUE_PROFILE) && defined(SANCOV_CMPLOG)
```

Comment:
These macros are a bit DRY; I should pull out the repeating stuff.

Reply:
it's ok

Reply:
Have tidied them up a bit now